app/js/users/userDetailController.js

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2015 Caleb Brose, Chris Fogerty, Rob Sheehy, Zach Taylor, Nick Miller
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+var _ = require('lodash');
+
+function userDetailController($scope, $routeParams, configService, flux, userModel, userService, beaconModel, beaconService) {
+    'use strict';
+
+    // init
+    $scope.user = { Email: $routeParams.email };
+    $scope.new = {
+        Email: $routeParams.email,
+        AuthLevel: '',
+        Password: '',
+        Beacons: {}
+    };
+
+    $scope.roles = configService.roles;
+    $scope.beacons = {};
+
+    // Form control / validation
+    $scope.submitting = false;
+    $scope.form = { submitted: false };
+
+    userService.getUser($scope.user.Email);
+    beaconService.getBeacons();
+
+    // State event listeners
+    $scope.$listenTo(userModel, function () {
+        $scope.user = userModel.getUser();
+        _.forEach(configService.roles, function (role) {
+            if ($scope.user.AuthLevel == role.AuthLevel) {
+                $scope.user.Role = role.DisplayName;
+            }
+        });
+        console.log($scope.user);
+    });
+
+    $scope.$listenTo(beaconModel, function () {
+        $scope.beacons = beaconModel.getBeacons();
+    });
+
+    $scope.open = function () {
+        $scope.submitting = true;
+    };
+
+    $scope.close = function () {
+        $scope.submitting = false;
+        $scope.form.submitted = false;
+        $scope.new.AuthLevel = '';
+        $scope.new.Password = '';
+        $scope.new.Beacons = {};
+    };
+
+    $scope.edit = function () {
+        if ($scope.userForm.$valid) {
+            var newUser = { Email: $scope.user.Email };
+            if ($scope.new.AuthLevel) {
+                newUser.AuthLevel = parseInt($scope.new.AuthLevel);
+            }
+            if ($scope.new.Password) {
+                newUser.Password = $scope.new.Password;
+            }
+            if ($scope.new.Beacons) {
+                newUser.Beacons = $scope.new.Beacons;
+            }
+            //userService.editUser($scope.user.Email, newUser);
+            console.log($scope.new);
```

Comment:
Yeah angular's scope binding can be a little weird sometimes. Especially with complex / nested objects. I can't figure out why this isn't working either. At first I thought it was because the `new.Beacons` object doesn't have the keys pre-populated so a radio click wouldn't actually update the model. I would suggest refactoring how you are storing the change. One option is to attach a scope function (via `ng-click`) to each radio button that passes the beacon and permission update as parameters and do the update there. Something like:

```
// authLevel here can be hardcoded for each radio value (-1, 0, 1, 2) in the repeat block
<input ng-click="modifyBeacon(beacon, authLevel)" ... >

$scope.modifyBeacon = function (beacon, authLevel) {
     $scope.new.Beacons[beacon.Address] = authLevel;
};
```

I also found an issue with the modal - it was not pre-populating my user's role, even though you were computing it in the `$scope.$listenTo(userModel, ...` block. Caleb's semantics make admins have a high authLevel value to have precedence over everything. In `admin@gmail.com`'s case, this value is 9001. Instead of explaining it here, I've made a [branch](https://github.com/lighthouse/harbor/compare/manage-users...manage-users-nick?expand=1)

Feel free to merge that in or modify as you wish. Let me know if any of that doesn't make sense :)